### DISCUSSION TITLE
JavaScript Function Issue

### DISCUSSION THREAD

**Discussion by Sachin2815** [microsoft/maker.js#606] [Q&A] [answered]
**Description:**
I am encountering an issue with a JavaScript function that is not producing the expected result. Here is the code I am using:

```javascript
function calculateSum(a, b) {
    let result = a + b;
    return result;
}

console.log(calculateSum(5, '10'));
 ```

  **What is causing this issue, and how can I modify the function to ensure it returns the correct sum when given a number and a string?**

**Comment by NitishKumar525** [comment, marked as answer]
## Solution

The issue arises due to JavaScript's type coercion rules. When the `+` operator is used with a number and a string, JavaScript converts the number to a string and performs string concatenation instead of numeric addition. 

To address this issue, you need to ensure that both parameters are treated as numbers before performing the addition. You can use the `Number` function to explicitly convert the string to a number:

```javascript
function calculateSum(a, b) {
    let result = Number(a) + Number(b);
    return result;
}

console.log(calculateSum(5, '10')); // Output: 15
